### PR TITLE
[SFSRAP-26] Fixed test not catching failures of the binding redirect.

### PR DESCRIPTION
**Pull request by HSAR**

### CODE REVIEW

File: Tests/ExampleAnalyzers/RoslynAnalyzer10/RoslynAnalyzer10/ExampleAnalyzer2.cs [hunk 1]
```diff
@@ -29,6 +29,9 @@ public class ExampleAnalyzer2 : DiagnosticAnalyzer
         private static readonly LocalizableString Description = "ExampleAnalyzer2 Description";
         private const string Category = "Naming";
 
+        // Referencing this forces some commonly-used libraries to be loaded
+        private static readonly LanguageVersion csVersion = Microsoft.CodeAnalysis.CSharp.LanguageVersion.CSharp1;
```
Comment (HSAR) [inline Tests/ExampleAnalyzers/RoslynAnalyzer10/RoslynAnalyzer10/ExampleAnalyzer2.cs]: We need to directly reference something inside a library in order to force it to load.

Previously we were importing Microsoft.CodeAnalysis.CSharp, but it wasn't being used.

File: Tests/IntegrationTests/IntegrationTests.csproj [hunk 2]
```diff
@@ -141,7 +141,9 @@
     </ProjectReference>
   </ItemGroup>
   <ItemGroup>
-    <None Include="App.config" />
+    <None Include="..\..\RoslynPluginGenerator\App.config">
+      <Link>App.config</Link>
+    </None>
```
Comment (HSAR) [inline Tests/IntegrationTests/IntegrationTests.csproj]: Changed duplicated files inside multiple projects to links back to the "true" product file, avoiding the possibility of getting out of sync.

**Comment by HSAR** [comment]
Microsoft.CodeAnalysis.CSharp.Workspaces and the other *Workspaces* libraries have nothing that can be referenced from outside.

File: Tests/IntegrationTests/IntegrationTests.csproj (hunk 2 above)
Comment (duncanpMS) [inline Tests/IntegrationTests/IntegrationTests.csproj]: Check that the old app.config files have been deleted; they don't appear to be removed in this commit.

**Comment by duncanpMS** [comment]
LGTM

**Pull request merged**